src/cudadev/DataFormats/SoAView.h

```diff
@@ -0,0 +1,375 @@
+/*
+ * Structure-of-Arrays template allowing access to a selection of scalars and columns from one
+ * or multiple SoAStores. A SoAStore is already a view to its complete set of columns.
+ * This class will allow handling subsets of columns or set of columns from multiple SoAViews, possibly
+ * with varying columns lengths.
+ */
+
+#ifndef DataStrcutures_SoAView_h
+#define DataStrcutures_SoAView_h
+
+#include "SoACommon.h"
+
+#define SoA_view_store(TYPE, NAME)                                                                                                  \
+  (TYPE, NAME)
+
+#define SoA_view_store_list(...)                                                                                                    \
+  __VA_ARGS__
+
+#define SoA_view_value(STORE_NAME, STORE_MEMBER, LOCAL_NAME)                                                                        \
+  (STORE_NAME, STORE_MEMBER, LOCAL_NAME)
+
+#define SoA_view_value_list(...)                                                                                                    \
+  __VA_ARGS__
+
+/*
+ * A macro defining a SoA view (collection of coluns from multiple stores)
+ * 
+ * Usage:
+ * generate_SoA_view(PixelXYView,
+ *   SoA_view_store_list(
+ *     SoA_view_store(PixelDigis,        pixelDigis),
+ *     SoA_view_store(PixelRecHitsStore, pixelsRecHit)
+ *   ),
+ *   SoA_view_value_list(
+ *     SoA_view_value(pixelDigis,   x,   digisX),
+ *     SoA_view_value(pixelDigis,   y,   digisY),
+ *     SoA_view_value(pixelsRecHit, x, recHitsX),
+ *     SoA_view_value(pixelsRecHit, y, recHitsY)
+ *   )
+ * );
+ *    
+ */
+
+/*
+ * Members definitions macros for viewa
+ */
+
+
+/**
+ * Store types aliasing for referencing by name
+ */
+#define _DECLARE_VIEW_STORE_TYPE_ALIAS_IMPL(TYPE, NAME)                                                                             \
+  typedef TYPE BOOST_PP_CAT(TypeOf_, NAME);
+
+#define _DECLARE_VIEW_STORE_TYPE_ALIAS(R, DATA, TYPE_NAME)                                                                          \
+  BOOST_PP_EXPAND(_DECLARE_VIEW_STORE_TYPE_ALIAS_IMPL TYPE_NAME)
+
+/**
+ * Member types aliasing for referencing by name
+ */
+#define _DECLARE_VIEW_MEMBER_TYPE_ALIAS_IMPL(STORE_NAME, STORE_MEMBER, LOCAL_NAME, DATA)                                            \
+  typedef BOOST_PP_CAT(TypeOf_, STORE_NAME) :: SoAMetadata:: BOOST_PP_CAT(TypeOf_, STORE_MEMBER) BOOST_PP_CAT(TypeOf_, LOCAL_NAME); \
+  static const SoAColumnType BOOST_PP_CAT(ColumnTypeOf_, LOCAL_NAME) =                                                              \
+    BOOST_PP_CAT(TypeOf_, STORE_NAME) :: SoAMetadata:: BOOST_PP_CAT(ColumnTypeOf_, STORE_MEMBER);                                   \
+  DATA BOOST_PP_CAT(TypeOf_, LOCAL_NAME) * BOOST_PP_CAT(addressOf_, LOCAL_NAME) () const  {                                         \
```

Comment:
Does this return a pointer-to-non-`const` from a `const` function? (I'm not really following the macro code)

Reply:
This macro makes the const/non-const selectable via the `DATA` parameter. It is either called with `BOOST_PP_EMPTY()` [in the mutable variant of SoAMetadata](https://github.com/ericcano/pixeltrack-standalone/blob/macroBasedSoA-PR211-followup/src/cudadev/DataFormats/SoAView.h#L322) or `const` [for the const variant](https://github.com/ericcano/pixeltrack-standalone/blob/macroBasedSoA-PR211-followup/src/cudadev/DataFormats/SoAView.h#L432).